OMCompiler/SimulationRuntime/c/simulation/solver/spatialDistribution.c

```diff
@@ -272,6 +274,22 @@ void storeSpatialDistribution(DATA* data, threadData_t *threadData, unsigned int
     omc_throw_function(threadData);
   }
 
+  /* Get deltaX */
+  deltaX = fabs(spatialDistribution->oldPosX - posX);
+  if (posX - spatialDistribution->oldPosX > 0) {
+    realDirection = 1 /* positive */;
+  } else if (posX - spatialDistribution->oldPosX < 0) {
+    realDirection = -1 /* negative */;
+  } else {
+    realDirection = 0 /* standing still */;
+  }
+
+  /* Check if real dirrection matches isPositiveVelocity */
+  if ((isPositiveVelocity && realDirection==-1 && deltaX > SPATIAL_ZERO_DELTA_X) || (!isPositiveVelocity && realDirection!= -1 && deltaX > SPATIAL_ZERO_DELTA_X)) {
+    // TODO: This is probably still a sign that we didn't handle some event or event search correctly.
+    isPositiveVelocity = isPositiveVelocity ? 0:1;
```

Comment:
I also do not understand this manipulation, you map (-1, 0, 1) to (0, 1, 0), but why?

Reply:
`isPositiveVelocity` represents a Boolean (so everything that is not `0` is true). We just flip it here.

You could write the same as:
```C
if (isPositiveVelocity != 0)
  isPositiveVelocity = 0
else 
  isPositiveVelocity  = 1
```

Reply:
Even better would be to do
```C
isPositiveVelocity  = !isPositiveVelocity;
``` I guess

Reply:
ah yes okay i totally was misled by the fact that realDirection is not a boolean. 

Nevermind i am good with this!

Reply:
> 
> 
> Even better would be to do
> 
> ```c
> isPositiveVelocity  = !isPositiveVelocity;
> ```

I guess that also applies to `isPositiveVelocity = isPositiveVelocity ? 0:1;`, that I found a bit odd at first sight :)